src/ufuncs.c

```diff
@@ -382,6 +382,125 @@ static void YY_b_p_func(char** args, npy_intp* dimensions, npy_intp* steps, void
 }
 static PyUFuncGenericFunction YY_b_p_funcs[1] = {&YY_b_p_func};
 
+/* Define the geom, X, Y -> bool functions (Ydd_b) prepared */
+static char GEOSContainsXY(void* context, void* g1, void* pg1, double x, double y) {
+  GEOSGeometry *geom = NULL;
+  const GEOSPreparedGeometry* prepared_geom_tmp = NULL;
+  char ret;
+
+#if !GEOS_SINCE_3_12_0
+  geom = create_point(context, x, y);
+  if (geom == NULL) {
+    return 2;
+  }
+#endif
+
+  char destroy_prepared = 0;
+  if (pg1 == NULL) {
+    prepared_geom_tmp = GEOSPrepare_r(context, g1);
+    if (prepared_geom_tmp == NULL) {
+      return 2;
```

Comment:
You'll need to destroy `geom` (`if !GEOS_SINCE_3_12_0`) here

Maybe easiest to firstdo the prepare logic? Then you can also merge the conditionals `if GEOS_SINCE_3_12_0`